### DISCUSSION TITLE
What exactly is ganache.provider() ?

### DISCUSSION THREAD

**Discussion by challet** [ConsenSys-archive/ganache#4109] [Q&A] [answered]
Hello, I may be confusing but is [ganache.provider()](https://github.com/trufflesuite/ganache#as-an-ethersjs-provider) an adapter to connect to a Ganache backend (cli ran for instance) or is it the actual EVM simulator running inside Node.js or the browser ?

**Comment by jeffsmale90** [comment, marked as answer]
`ganache.provider()` will create an instance of Ganache (within the javascript context; nodejs or browser), and return an EIP-1193 provider to that Ganache instance. You can configure the Ganache instance by providing startup options object.

```javascript
const ganache = require("ganache");

const options = {};
const provider = ganache.provider(options);
const accounts = await provider.request({ method: "eth_accounts", params: [] });
```

If you wanted to have a provider to an instance of Ganache running in the CLI, you could use web3js, ethers or other library, and pass in the websockets or http url for your CLI instance of Ganache.

Hope that helps!

**Reply by challet** [reply]
Thank you for the clear answer. I was indeed messing things by running a Ganache CLI and trying to connect it with `ganache.provider()`
